src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -529,10 +537,43 @@ namespace winrt::TerminalApp::implementation
     // - <none>
     // Return Value:
     // - <none>
-    void TerminalPage::_CompleteInitialization()
+    winrt::fire_and_forget TerminalPage::_CompleteInitialization()
     {
         _startupState = StartupState::Initialized;
-        _InitializedHandlers(*this, nullptr);
+
+        // GH#632 - It's possible that the user tried to create the terminal
+        // with only one tab, with only an elevated profile. If that happens,
+        // we'll create _another_ process to host the elevated version of that
+        // profile. This can happen from the jumplist, or if the default profile
+        // is `elevate:true`, or from the commandline.
+        //
+        // However, we need to make sure to close this window in that scenario.
+        // Since there aren't any _tabs_ in this window, we won't ever get a
+        // closed event. So do it manually.
+        auto weakThis{ get_weak() };
+        if (_tabs.Size() == 0)
+        {
+            // This is MENTAL. If we exit right away after spawning the elevated
+            // WT, then ShellExecute might not successfully complete the
+            // elevation. What's even more, the Terminal will mysteriously crash
+            // somewhere in XAML land.
+            //
+            // So I'm introducing a 5s delay here for the Shell to complete the
+            // execution, and _then_ close the window.
+            //
+            // TODO! There's no way this is the right answer, right?
```

Comment:
TODO here.

Reply:
Yea so this is wild. I hate this 5s timeout and I totally forgot it was there (admittedly wrote that like 6months ago)

I'm trying out different ShellExecuteEx flags and non work the way we want. What we really want it just to spawn something elevated, fire-and-forget style. Problem is, ShellExecute seemingly requires that the caller is still alive when the UAC dialog is confirmed. If I remove the timeout, then non of `SEE_MASK_ASYNCOK`, `SEE_MASK_DEFAULT`, nor `SEE_MASK_NOASYNC` seem to work - the source wt always exits immediately, and the elevated one never opens. 

UNLESS ANYONE HAS A BETTER IDEA, I'm gonna write another shim `elevate-helper.exe` that's literally just `ShellExecute("runas", args[1:])`, and `CreateProcess` that helper instead. That will get the ShellExecute out of WT's way, and allow SE to wait on the helper rather than us. 

Oh unless this is a breakaway thing. Hmm.